lms/djangoapps/course_api/blocks/transformers/milestones.py

```diff
@@ -172,3 +168,40 @@ def gated_by_required_content(block_key, block_structure, required_content):  #
             return True
 
         return False
+
+    def _generate_special_exam_attempt_context(self, block_key, block_structure, usage_info):
+        """
+        Helper method which generates the special exam attempt context.
+        Either calls into proctoring or, if exams ida waffle flag on, then get internally.
+        """
+        special_exam_attempt_context = None
+
+        # if exams waffle flag enabled, get exam type internally
+        if exams_ida_enabled(block_key.course_key):
+            # add short description based on exam type
+            if block_structure.get_xblock_field(block_key, 'is_practice_exam'):
+                exam_type = _('Practice Exam')
+            elif block_structure.get_xblock_field(block_key, 'is_proctored_enabled'):
+                exam_type = _('Proctored Exam')
+            elif block_structure.get_xblock_field(block_key, 'is_timed_exam'):
+                exam_type = _('Timed Exam')
+            else:  # sets a default, though considered impossible
+                log.info('Using a default value, but it is considered impossible.')
```

Comment:
default value for what? Maybe "using default 'exam' value for exam type" or similar so there's a hint about where to look